### PR TITLE
Upgrade jitsi

### PR DESCRIPTION
**Pull request by bmarty**
Upgrade Jitsi lib from 2.9.3 to 3.1.0 and drop support for Android 5.0.

Also fix bug about avatar Url and improve code regarding the $theme pattern

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/widget/RoomWidgetItem.kt [hunk 1]
```diff
@@ -43,7 +43,7 @@ abstract class RoomWidgetItem : EpoxyModelWithHolder<RoomWidgetItem.Holder>() {
     override fun bind(holder: Holder) {
         super.bind(holder)
         holder.widgetName.text = widget.name
-        holder.widgetUrl.text = tryOrNull { URL(widget.computedUrl) }?.host ?: widget.computedUrl
+        holder.widgetUrl.text = tryOrNull { URL(widget.widgetContent.url) }?.host ?: widget.widgetContent.url
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/home/room/detail/widget/RoomWidgetItem.kt:46]: Why are you using url instead of computed?

**Review by ganfra — COMMENTED** [review]
Some questions/remarks

File: vector/src/main/java/im/vector/app/features/home/room/detail/widget/RoomWidgetItem.kt (hunk 1 above)
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/home/room/detail/widget/RoomWidgetItem.kt:46]: We just need the `host` here. OK for you?

File: vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetProperties.kt [hunk 2]
```diff
@@ -19,22 +19,32 @@ package im.vector.app.features.call.conference
...
-    val confId: String? by lazy { configs["conferenceId"] }
-    val displayName: String? by lazy { configs["displayName"] }
-    val avatarUrl: String? by lazy { configs["avatarUrl"] }
+data class JitsiWidgetProperties(
+        val domain: String,
+        val confId: String?,
+        val displayName: String?,
+        val avatarUrl: String?
+) {
+    companion object {
```
[the remark this answers is not in the document]
Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/call/conference/JitsiWidgetProperties.kt:31]: Good idea, done

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/widgets/WidgetManager.kt [hunk 3]
```diff
@@ -104,6 +104,10 @@ internal class WidgetManager @Inject constructor(private val integrationManager:
         return widgetEvents.mapEventsToWidgets(widgetTypes, excludedTypes)
     }
 
+    fun getWidgetComputedUrl(widget: Widget, isLightTheme: Boolean): String? {
```
[the remark this answers is not in the document]
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/widgets/WidgetManager.kt:107]: I do not change it for now

**Review by ganfra — APPROVED** [review]
LGTM, thanks!

**Pull request merged**